src/lib.rs

```diff
@@ -166,11 +182,17 @@ impl DesktopEntry {
         // If appid matches
         id == self.id()
             // or startup_wm_class matches
-            || self.startup_wm_class().map_or(false, |wm_class| wm_class == id)
+            || self.startup_wm_class().is_some_and(|wm_class| wm_class == id)
+            // or X-Ubuntu-Gettext-Domain was set
+            || self.ubuntu_gettext_domain.as_ref().is_some_and(|domain| *domain == id)
```

Comment:
Does documentation on `X-Ubuntu-Gettext-Domain` exist? I wasn't able to find anything official and it seems to be related to translations, so I would argue, also not in spec.

Reply:
Works for `seahorse`, but maybe there's a better way. The desktop entry is named `org.gnome.seahorse.Application` and the only exact match for the ID is either `X-Ubuntu-Gettext-Domain=seahorse` or `Exec=seahorse %u`.

Reply:
Found that too and the app id seems to indeed by "seahorse" (or "Seahorse" if you use the X11 backend), so how the heck is this working?

Reply:
Seems this was recently fixed upstream at least: https://gitlab.gnome.org/GNOME/seahorse/-/merge_requests/238